br/pkg/lightning/mydump/csv_parser.go

```diff
@@ -370,11 +378,43 @@ func (parser *CSVParser) readRecord(dst []string) ([]string, error) {
 
 	isEmptyLine := true
 	whitespaceLine := true
+	foundStartingByThisLine := false
 	prevToken := csvTokenNewLine
 	var firstToken csvToken
 
 outside:
 	for {
+		// we should drop
+		// 1. the whole line if it does not contain startingBy
+		// 2. any character before startingBy
+		// since we have checked startingBy does not contain terminator, we can
+		// split at terminator to check the substring contains startingBy. Even
+		// if the terminator is inside a quoted field which means it's not the
+		// end of a line, the substring can still be dropped by rule 2.
+		if len(parser.startingBy) > 0 && !foundStartingByThisLine {
```

Comment:
```suggestion
		for len(parser.startingBy) > 0 {
```

Reply:
Nit: maybe more readable

Reply:
when the `outside` for loop goes to next round, it does not mean a line is processed (it only means we have met a special token). So we need another variable to record the state per line.